feat: add OpenAI GPT-5.1 series models

PeterDaveHello opened the pull request:
### **User description**
- add support for gpt-5.1, gpt-5.1-2025-11-13, gpt-5.1-chat-latest, gpt-5.1-codex, gpt-5.1-codex-mini
- replace default model gpt-5-2025-08-07 with gpt-5.1-2025-11-13 per OpenAI’s Nov 13 launch

See: https://openai.com/index/gpt-5-1-for-developers/

GitHub Copilot Summary:

> This pull request updates the default model configuration to use a newer version and expands support for additional GPT-5.1 model variants. The most important changes are:
> 
> **Model Support Updates:**
> 
> * Added support for several new GPT-5.1 model variants in the `pr_agent/algo/__init__.py` file, including `gpt-5.1`, `gpt-5.1-2025-11-13`, `gpt-5.1-chat-latest`, `gpt-5.1-codex`, and `gpt-5.1-codex-mini`, all with a token limit of 200,000.
> 
> **Configuration Changes:**
> 
> * Updated the default `model` in `pr_agent/settings/configuration.toml` from `gpt-5-2025-08-07` to the newer `gpt-5.1-2025-11-13`.


___

### **PR Type**
Enhancement


___

### **Description**
- Add support for five new GPT-5.1 model variants

- Update default model to gpt-5.1-2025-11-13

- All new models configured with 200K token limit


___

### Diagram Walkthrough


```mermaid
flowchart LR
  A["Model Registry"] -->|"Add 5 variants"| B["GPT-5.1 Models"]
  C["Default Config"] -->|"Update to latest"| D["gpt-5.1-2025-11-13"]
  B --> E["200K tokens each"]
```



<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Register GPT-5.1 model variants</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/algo/__init__.py

<ul><li>Added five new GPT-5.1 model variants to the model registry<br> <li> Models include <code>gpt-5.1</code>, <code>gpt-5.1-2025-11-13</code>, <code>gpt-5.1-chat-latest</code>, <br><code>gpt-5.1-codex</code>, and <code>gpt-5.1-codex-mini</code><br> <li> All variants configured with 200,000 token limit</ul>


</details>


  </td>
  <td><a href="https://github.com/qodo-ai/pr-agent/pull/2103/files#diff-5923c546f24ec7308a0e43fc84bb6fe40de7bfe2ac6ee842da9578e5dc2c692b">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>configuration.toml</strong><dd><code>Update default model to GPT-5.1</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/settings/configuration.toml

<ul><li>Updated default model from <code>gpt-5-2025-08-07</code> to <code>gpt-5.1-2025-11-13</code><br> <li> Aligns with OpenAI's November 13 GPT-5.1 launch</ul>


</details>


  </td>
  <td><a href="https://github.com/qodo-ai/pr-agent/pull/2103/files#diff-66cfda5143e484ee53ecf7aa0df7dca8ad0b181256f4b0675905db35bcbbae78">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

ofir-frd commented on pr_agent/settings/configuration.toml:
```diff
@@ -6,7 +6,7 @@
 [config]
 # models
-model="gpt-5-2025-08-07"
+model="gpt-5.1-2025-11-13"
```
Our soon-to-be-published benchmarks indicate that GPT-5.1 does not outperform GPT-5.

ofir-frd reviewed (changes requested):
To approve this, please revert the default model config and bump the LiteLLM version.

Thanks!

PeterDaveHello committed:
feat: add OpenAI GPT-5.1 series models

- add support for gpt-5.1, gpt-5.1-2025-11-13, gpt-5.1-chat-latest, gpt-5.1-codex, gpt-5.1-codex-mini
- replace default model gpt-5-2025-08-07 with gpt-5.1-2025-11-13 per OpenAI’s Nov 13 launch

See: https://openai.com/index/gpt-5-1-for-developers/

ofir-frd reviewed (approved):
Approved, thank you for the contribution!
